src/kirby/components/dropdown/dropdown.component.ts

```diff
@@ -0,0 +1,256 @@
+          this.onItemSelect(index);
+        });
+      });
+    }
+  }
+
+  open() {
+    if (!this.isOpen) {
+      this._isOpening = true;
+      setTimeout(() => {
+        this.isOpen = true;
+        this.scrollItemIntoView(this.selectedIndex);
+      });
+    }
+  }
+
+  close() {
+    if (this.isOpen) {
+      this._isOpening = false;
+      this.isOpen = false;
+    }
+  }
+
+  onItemSelect(index: number) {
+    this.selectItem(index);
+    this.close();
+  }
+
+  private selectItem(index: number) {
+    this.selectedIndex = index;
+    this.selectedItem = this.items[index];
+    this.setSelectedText();
+    this.scrollItemIntoView(index);
+  }
+
+  private setSelectedText() {
+    let selectedText = this.placeholder;
+    if (this.selectedItem) {
+      selectedText = this.getTextFromItem(this.selectedItem);
+    }
+    this.selectedText = selectedText;
+  }
+
+  getTextFromItem(item: string | any) {
+    return typeof item === 'string' ? item : item[this.itemTextProperty];
+  }
+
+  scrollItemIntoView(index: number) {
+    const kirbyItems = this.kirbyItemsSlotted.length
+      ? this.kirbyItemsSlotted
+      : this.kirbyItemsDefault;
+    if (kirbyItems.length) {
+      const selectedKirbyItem = kirbyItems.toArray()[index];
+      if (selectedKirbyItem && selectedKirbyItem.nativeElement) {
+        const itemElement = selectedKirbyItem.nativeElement;
+        const scrollContainer = this.cardElement.nativeElement;
+        const itemTop = itemElement.offsetTop;
+        const itemBottom = itemElement.offsetTop + itemElement.offsetHeight;
+        const containerVisibleTop = scrollContainer.scrollTop;
+        const containerVisibleBottom = scrollContainer.clientHeight + scrollContainer.scrollTop;
+        if (itemTop < containerVisibleTop) {
+          scrollContainer.scrollTop = itemTop;
+        } else if (itemBottom > containerVisibleBottom) {
+          scrollContainer.scrollTop = itemBottom - scrollContainer.clientHeight;
+        }
+      }
+    }
+  }
+
+  @HostListener('keydown.tab', ['$event'])
+  private _onTab(event: KeyboardEvent) {
+    if (this.isOpen) {
+      event.preventDefault();
+      this.close();
+    }
+  }
+
+  @HostListener('keydown.enter')
+  @HostListener('keydown.escape')
+  @HostListener('blur')
+  private _onBlur() {
+    if (this.isOpen) {
+      this.close();
+    }
+  }
+
+  @HostListener('keydown.space', ['$event'])
+  private _onSpace(event: KeyboardEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+    if (!this.isOpen) {
+      this.open();
+    }
+  }
+
+  @HostListener('keydown.enter', ['$event'])
+  private _onEnter(event: KeyboardEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+    this.toggle();
+  }
+
+  @HostListener('keydown.arrowup', ['$event'])
+  @HostListener('keydown.arrowdown', ['$event'])
+  @HostListener('keydown.arrowleft', ['$event'])
+  @HostListener('keydown.arrowright', ['$event'])
+  private _onArrowKeys(event: KeyboardEvent) {
```

Comment:
It's not possible to do AOT-compile `@HostBindings` to private functions - which is a requisite for producing an transpiled version of kirby. Instead - consider making the function public (but keep the underscore prefix to indicate that it should be consider `private`)?